include/rppt_tensor_statistical_operations.h

```diff
@@ -267,6 +267,54 @@ RppStatus rppt_tensor_stddev_host(RppPtr_t srcPtr, RpptDescPtr srcDescPtr, RppPt
 RppStatus rppt_tensor_stddev_gpu(RppPtr_t srcPtr, RpptDescPtr srcDescPtr, RppPtr_t tensorStddevArr, Rpp32u tensorStddevArrLength, Rpp32f *meanTensor, RpptROIPtr roiTensorPtrSrc, RpptRoiType roiType, rppHandle_t rppHandle);
 #endif // GPU_SUPPORT
 
+/*! \brief Threshold augmentation on HOST backend for a NCHW/NHWC layout tensor
+ * \details The Threshold augmentation applies min, max threshold checks for a batch of RGB(3 channel) / greyscale(1 channel) images with an NHWC/NCHW tensor layout.<br>
+ * - srcPtr depth ranges - Rpp8u (0 to 255), Rpp16f (0 to 1), Rpp32f (0 to 1), Rpp8s (-128 to 127).
+ * - dstPtr depth ranges - Will be same depth as srcPtr.
+ * \image html img150x150.png Sample Input
+ * \image html statistical_operations_threshold_img150x150.png Sample Output
+ * \param [in] srcPtr source tensor in HOST memory
+ * \param [in] srcDescPtr source tensor descriptor (Restrictions - numDims = 4, offsetInBytes >= 0, dataType = U8/F16/F32/I8, layout = NCHW/NHWC, c = 1/3)
+ * \param [out] dstPtr destination tensor in HOST memory
+ * \param [in] dstDescPtr destination tensor descriptor (Restrictions - numDims = 4, offsetInBytes >= 0, dataType = U8/F16/F32/I8, layout = NCHW/NHWC, c = same as that of srcDescPtr)
+ * \param [in] minTensor minimum pixel value (1D tensor in HOST memory, of size batchSize * channels)
+ * - minTensor ranges - Rpp8u (0 to 255), Rpp16f (0 to 1), Rpp32f (0 to 1), Rpp8s (0 to 255).
+ * \param [in] maxTensor maximum pixel value (1D tensor in HOST memory, of size batchSize * channels)
+ * - maxTensor ranges - Rpp8u (0 to 255), Rpp16f (0 to 1), Rpp32f (0 to 1), Rpp8s (0 to 255).
```

Comment:
Internally we are doing a addition of +128 for all the I8 load functions
https://github.com/sampath1117/rpp/blob/sr/opt_threshold/src/include/cpu/rpp_cpu_simd.hpp#L572

So if we use a equivalent I8 range threshold for min and max
It will result in different outputs compared to U8 since we are anyway converting the input to U8 range in above load functions

Reply:
Modified the code now to use threshold in I8 range itself as is:
https://github.com/sampath1117/rpp/commit/658acaafa5cd5b33546c6d4d11d478e24fbdba38